libzfs-sys/src/lib.rs

```diff
@@ -241,31 +265,25 @@ mod tests {
 
             let mut pl_p = prop_list_ptr;
 
-            fn create_with_capacity(x: c_uint) -> *mut c_char {
-                let s = String::with_capacity(x as usize);
-                let c_string = CString::new(s).unwrap();
-                c_string.into_raw()
-            }
-
             while !pl_p.is_null() {
                 let zfs_prop = to_zfs_prop_t((*pl_p).pl_prop).unwrap();
 
                 if zfs_prop != zfs_prop_t::ZFS_PROP_BAD {
-                    let raw = create_with_capacity(ZFS_MAXPROPLEN);
-                    let source_raw = create_with_capacity(ZFS_MAX_DATASET_NAME_LEN);
-                    let source_t = ptr::null_mut();
+                    let raw = CString::new("0".repeat(319)).unwrap().into_raw();
```

Comment:
What is 319?

Reply:
320 is the maximum size I found usable for the pre-allocated buffer that gets the property.

319 + null byte = 320.